lib/dock.rb

```diff
@@ -0,0 +1,58 @@
+class Dock
+  attr_reader :name, :max_rental_time
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = 3
+    @revenue = 0
+    @boats = {}
+
+  end
+
+  def rent(boat,renter)
+    @boats[boat] = renter
+  end
+
+  def log_hour
+    @boats.each do |boat, renter|
+        boat.add_hour
+      end
+  end
+
+  def return(boat)
+    boat
+    # @boats.each do |boat, rented|
+    #   boat.hours_rented = 0
+    #   binding.pry
+    # end
+  end
+
+  def revenue
```

Comment:
I looked at the working version from a couple commits ago - that looks great!  One thing I would recommend is to use `each` instead of map, so that you can return `@revenue` at the end of the method - it is a bit clearer as to what you are doing :)